src/libstd/keyword_docs.rs

```diff
@@ -925,8 +925,81 @@ mod mut_keyword {}
 //
 /// Make an item visible to others.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// The keyword `pub` makes any module, function, or data structure accessible from inside
+/// of external modules. The `pub` keyword may also be used in a `use` declaration to re-export
+/// an identifier from a namespace.
+///
+///  In the below example we define `PublicStruct` within `public_inner_module` and make both public
+///  using the `pub` keyword. We write `pub use` in `secondary_mod` to re-export `PublicStruct`
+/// and we use `use secondary_mod::PublicStruct` in our `main` function to import `PublicStruct` as if
+/// `PublicStruct` were local to `secondary_mod`.
+///
+/// ```rust
+/// mod primary_mod {
+///    // Our inner module and struct we've made public
+///    pub mod public_inner_module {
+///             pub struct PublicStruct;
+///     }
+///     # fn main() {
+///     // Using our public struct locally.
+///     let localStruct = public_inner_module::PublicStruct;
+///     # }
+/// }
+///
+/// mod secondary_mod {
+///     // We import PublicStruct into secondary_mod's namespace with our use declaration
+///     // and re-export it by prefixing our use declaration with pub.
+///     pub use crate::primary_mod::public_inner_module::PublicStruct;
+///
+///     // using our PublicStruct from within our secondary module.
+///     # fn main() {
+///     let foreignStruct = PublicStruct;
+///     # }
+/// }
+///
+/// fn main() {
+///     // When we import secondary_mod we may use PublicStruct
+///     // as if it were defined in secondary_mod.
+///     use secondary_mod::PublicStruct;
+///
+///
+///     let emptyStruct = PublicStruct;
+/// }
+/// ```
+///
+/// In the below example, private_module does not use the `pub` keyword and therefore
+/// `HiddenStruct`, despite it's usage of the `pub` keyword is inaccessible.
+///
+///
+/// ```compile_fail
+/// mod other_mod {
+///     mod private_module {
+///         pub struct HiddenStruct;
+///     }
+/// }
+///
+/// // We can't initialize emptyStruct because we don't have access to HiddenStruct.
+/// let emptyStruct = other_mod::private_module::HiddenStruct;
+/// ```
+///
+/// Likewise, code inside of a public module, is inaccessible
+/// unless it uses the `pub` keyword.
+///
+/// ```compile_fail
+/// mod other_mod {
+///     pub mod public_module {
+///         fn private_function() -> () {}
+///     }
+/// }
+///
+/// // We can't use private function because it's private.
+/// private_function();
+/// ```
 ///
+/// For more information on the `pub` keyword, please see the [module] section
```

Comment:
I think linking to https://doc.rust-lang.org/reference/visibility-and-privacy.html is a better fit here, it seems like the most exhaustive and best documentation we have right now.